docs/Account/methods.md

```diff
@@ -0,0 +1,284 @@
+:::
+
+## Smart Account Get Methods
+
+### getAccountAddress( )
+
+This method is used to retrieve the account address associated with the `smartAccount` instance.
+
+**Usage**
+
+```jsx
+const address = await smartAccount.getAccountAddress();
+```
+
+**Returns**
+
+- address (`Promise<string>`): A Promise resolving to the account address associated with the smartAccount instance.
+
+### getNonce( )
+
+This method is used to retrieve the nonce associated with the `smartAccount` instance.
+
+**Usage**
+
+```jsx
+
+const nonce = await smartAccount.getNonce();
+console.log(nonce.toNumber());
+```
+**Parameters**
+- nonceKey(`number`): one can also pass the optional parameter nonceKey in the case of two-dimensional nonces
+
+**Returns**
+
+- address (`Promise<BigNumber>`): A Promise resolving to BigNumber indicating the nonce of the smart account.
+
+### index
+
+This is used to retrieve the index of the current active smart account.
+
+**Usage**
+```jsx
+const index = smartAccount.index;
+```
+
+**Returns**
+- index (`number`): A number indicating the index of current active smart account.
+
+## UserOp Methods
+
+### buildUserOp( )
+This method is used for configuring and setting up all properties of the `userOp` object.  It involves preparing and populating the userOp with necessary information and parameters required for execution by bundler.
+
+**Usage**
+
+For example, in the context of creating a userOp for an `addComment` transaction, an instance of the contract is created, then a basic transaction object is created that holds the necessary address and data from the transaction. Finally, a userOp is created using the Smart Account's `buildUserOp` method⁠.
+
+```jsx
+const contractAddress = "contract address";
+const provider = new ethers.providers.JsonRpcProvider("rpc url");
+
+const blogContract = new ethers.Contract(
+  contractAddress,
+  abi, // contract abi
+  provider,
+);
+
+const createComment =
+  await blogContract.populateTransaction.addComment("comment");
+
+const tx1 = {
+  to: contractAddress,
+  data: createComment.data,
+};
+
+const userOp = await smartAccount.buildUserOp([tx1]);
+```
+
+**Parameters**
+- transactions (`Transaction[]`, required): The required argument  is an array of transactions. You can pass multiple transactions into a userOp if you would like to batch them together into one transaction.
+- buildUseropDto (`BuildUserOpOptions`): One can also pass these options to customize how a userOp is built.
+
+  ```ts
+  type BuildUserOpOptions = {
+    overrides?: Overrides;
+    skipBundlerGasEstimation?: boolean;
+    params?: ModuleInfo;
+    nonceOptions?: NonceOptions; 
+    forceEncodeForBatch?: boolean;
+    paymasterServiceData?: SponsorUserOperationDto;
+  }
+  ```
+  Let's look at each of these params:
+  1. overrides (`Overrides`): one can override any of the values of the userOp when it is being constructed.
```

Comment:
This does not work the way it is explained. 
In the code if you notice it is only overridden for estimate gas request to the bundler.

Reply:
![Uploading Screenshot 2024-01-03 at 10.24.50 PM.png…]()

Reply:
cant see the screenshot

Reply:
![Uploading Screenshot 2024-01-03 at 10.24.50 PM.png…]()

Reply:
https://imgur.com/a/gCG6nC3

Reply:
like the overrides would only be used for input to the bundler to do the gas estimations (eth_estimateUserOpGas call) and returned values from the bundler will be used (so these don't act as direct overrides)

but before signing userop or before sending to paymaster service one can directly replace gas limits in partial userop if need to